### PR TITLE
adds jars and saline

### PR DESCRIPTION
**Pull request by MadmanMartian**
Literally fluff/cosmetic stuff.

Jars can be used to hold something in a solution of something (Think mini-cryo) where whatever reagents are in the jar act on the item contained within.

This does mean that you can fill it with polyacid and dip things in, assuming you don't do this.

welp

Should you put an organ object into it and the jar contains clonexadone, it will 'heal' the organ. This can not undo organ death however.

## Misc changes
Redoes the description on how to calculate density and SHC at the top of chemistry-reagents.dm

Still prodding and poking. Not done as of yet.

- [x] Add way to acquire a Jar.
- [x] Beautify the process.

:cl:
* rscadd: Adds jars for storing reagents and items in. Items in the jar are constantly touched by the reagents inside, so be careful what you mix up.
* rscadd: Internal organs in the presence of peridaxon are now healed over time.

### CODE REVIEW

**Comment by SonixApache** [comment]
does saline work as a shitty alternative to blood?

**Comment by MadmanMartian** [comment]
Nope. That'd be unatomic. Right now it's just an organ preservative.

**Comment by Pathid** [comment]
If the preservative is called saline then inevitably some players are going to think it has a medical use in the body.

**Comment by Hugh-G-Rection** [comment]
Feature suggestion: You can throw jars, breaking it and releasing it's contents to splash chemicals on things and people.
Suggestion 2: You can use a net to add bees to a jar, creating an angry bee grenade.

**Comment by Diavill** [comment]
pickles when

**Comment by MadmanMartian** [comment]
@Hugh-G-Rection prior already in, latter does sound fun however

**Comment by Dacendeth** [comment]
Any reaction if it fill it with welding fluid and put a zippo/litflare/etc inside?

**Comment by MadmanMartian** [comment]
none, but that would make for an interesting molotov if you throw it at somebody

**Comment by Kurfursten** [comment]
Mysteriously has nothing to do with /obj/item/weapon/reagent_containers/food/drinks/jar

**Comment by MadmanMartian** [comment]
that thing is something strange that I am intending to avoid

File: code/modules/reagents/reagent_containers/glass/jar.dm [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	w_class = W_CLASS_MEDIUM
+	melt_temperature = MELTPOINT_GLASS
+	origin_tech = Tc_MATERIALS + "=1"
+	var/obj/held_item = null
+
+/obj/item/weapon/reagent_containers/glass/jar/New()
+	..()
+	processing_objects.Add(src)
+
+/obj/item/weapon/reagent_containers/glass/jar/Destroy()
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/glass/jar.dm:21]: You need to remove it from processing objects.

File: code/modules/reagents/reagent_containers/glass/jar.dm [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+				filling.icon_state = "[icon_state]75"
+			else
+				filling.icon_state = "[icon_state]100"
+
+		filling.icon += mix_color_from_reagents(reagents.reagent_list)
+		filling.alpha = mix_alpha_from_reagents(reagents.reagent_list)
+		overlays += filling
+
+	if(held_item)
+		var/image/contained_within = image("icon"=held_item)
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/glass/jar.dm]: Use mutable appearances here instead.

File: code/modules/reagents/reagent_containers/glass/jar.dm [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		underlays += contained_within
+
+/obj/item/weapon/reagent_containers/glass/jar/attackby(obj/item/I, mob/user, params)
+	..()
+	if(I.w_class <= w_class && !held_item)
+		if(user.drop_item(I, src))
+			to_chat(user, "<span class = 'notice'>You place \the [I] into \the [src]</span>")
+			held_item = I
+			update_icon()
+			return
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/glass/jar.dm]: IT TRAILS

File: code/modules/reagents/reagent_containers/glass/jar.dm [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+/obj/item/weapon/reagent_containers/glass/jar/process()
+	if(!held_item)
+		return
+
+	if(held_item.gcDestroyed)
+		held_item = null
+		update_icon()
+		return
+
+	if(istype(held_item, /obj/item/organ/internal))
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/glass/jar.dm]: Yeeeaaah this hardcode's not passing sorry.

**Review by PJB3005 — DISMISSED** [review]

File: code/modules/reagents/reagent_containers/glass/jar.dm (hunk 4 above)
Comment (MadmanMartian) [reply code/modules/reagents/reagent_containers/glass/jar.dm]: what would you suggest then?

**Comment by Blithering** [comment]
Now we just need someway to keep a head alive in a jar

**Comment by SonixApache** [comment]
@Blithering just stick an MMI in a jar.

File: code/modules/reagents/reagent_containers/glass/jar.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/modules/reagents/reagent_containers/glass/jar.dm:21]: removed from processing objects

**Comment by MadmanMartian** [comment]
Will conflict with #17529 both in file, and in recipe. I'm opting to be merged after it so I can change my recipe to require the saltwater reagent instead.

**Review by jknpj — APPROVED** [review]

**Pull request merged**